File: ABC347/c.py
# これはTLE
# n, a, b = map(int, input().split())

# d = list(map(int, input().split()))

# for today in range(1, 1 + a + b):
#     is_all_day_valid = True
#     for event_day in d:
#         if not(1 <= (today + event_day) % (a + b) <= a):
#             is_all_day_valid = False
#             break
#     if is_all_day_valid:
#         print("Yes")
#         exit()
# print("No")


N, A, B = map(int, input().split())
D = [int(i) % (A + B) for i in input().split()]
D = sorted(set(D))

if len(D) == 1:
    print('Yes')
else:
    for i in range(len(D)):
        if (D[(i + 1) % len(D)] - D[i]) % (A + B) > B:
            print('Yes')
            break
    else:
        print('No')

File: ABC347/c.cpp
// 何かの間違いでC++なら通ることに賭けてみる
// -> TLE
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, a, b;
    cin >> n >> a >> b;

    vector<int> d(n);
    for (int i = 0; i < n; ++i) {
        cin >> d[i];
    }

    for (int today = 1; today <= a + b; ++today) {
        bool is_all_day_valid = true;
        for (int event_day : d) {
            if (!((1 <= (today + event_day) % (a + b)) && ((today + event_day) % (a + b) <= a))) {
                is_all_day_valid = false;
                break;
            }
        }
        if (is_all_day_valid) {
            cout << "Yes" << endl;
            return 0;
        }
    }
    cout << "No" << endl;
    return 0;
}
